File: android/app/src/main/java/com/brokenlunch/gr/data/model/Submission.kt
package com.brokenlunch.gr.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class SubmissionRequest(
    @Json(name = "restaurant_id") val restaurantId: String,
    @Json(name = "menu_name") val menuName: String,
    @Json(name = "price_cents") val priceCents: Int,
    @Json(name = "photo_url") val photoUrl: String? = null,
    @Json(name = "gemini_parsed") val geminiParsed: GeminiParsedInfo? = null,
    @Json(name = "source") val source: String,
)

@JsonClass(generateAdapter = true)
data class GeminiParsedInfo(
    @Json(name = "confidence") val confidence: Double,
    @Json(name = "raw_text") val rawText: String? = null,
)

@JsonClass(generateAdapter = true)
data class SubmissionResponse(
    @Json(name = "id") val id: String,
    @Json(name = "menu_item_id") val menuItemId: String,
    @Json(name = "status") val status: String,
    @Json(name = "points_awarded") val pointsAwarded: Int,
    @Json(name = "is_first_submission") val isFirstSubmission: Boolean,
    @Json(name = "bonus_message") val bonusMessage: String? = null,
    @Json(name = "user_total_points") val userTotalPoints: Int,
    @Json(name = "user_level") val userLevel: Int,
    @Json(name = "level_up") val levelUp: Boolean,
)

File: backend/app/schemas/submission.py
"""Request/response models for POST /submissions."""

from __future__ import annotations

from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

SubmissionSource = Literal[
    "gemini_photo", "gemini_web", "user_manual", "seed", "places_api"
]


class SubmissionCreate(BaseModel):
    model_config = ConfigDict(extra="ignore")

    restaurant_id: UUID
    menu_name: str = Field(min_length=1, max_length=255)
    price_cents: int = Field(ge=1, le=1500)
    photo_url: str | None = None
    gemini_parsed: dict[str, Any] | None = None
    source: SubmissionSource = "gemini_photo"


class SubmissionResponse(BaseModel):
    id: UUID
    menu_item_id: UUID
    status: str
    points_awarded: int
    is_first_submission: bool
    bonus_message: str | None = None
    user_total_points: int
    user_level: int
    level_up: bool
